extractors/extractors.go

```diff
@@ -0,0 +1,511 @@
+//	    extractors.FromHeader("X-API-Key"),
+//	    extractors.FromCookie("api_key"),
+//	    extractors.FromQuery("token"),
+//	)
+//
+// Security considerations:
+//   - Query parameters and form data can leak sensitive information
+//   - Use HTTPS to protect extracted values in transit
+//   - Consider source-specific security policies for your use case
+
+import (
+	"errors"
+	"net/url"
+
+	"github.com/gofiber/fiber/v3"
+	utils "github.com/gofiber/utils/v2"
+)
+
+// Source represents the type of source from which a value is extracted.
+// This is informational metadata that helps developers understand the extractor behavior.
+type Source int
+
+const (
+	// SourceHeader indicates the value is extracted from an HTTP header.
+	SourceHeader Source = iota
+
+	// SourceAuthHeader indicates the value is extracted from the Authorization header.
+	SourceAuthHeader
+
+	// SourceForm indicates the value is extracted from form data.
+	SourceForm
+
+	// SourceQuery indicates the value is extracted from URL query parameters.
+	SourceQuery
+
+	// SourceParam indicates the value is extracted from URL path parameters.
+	SourceParam
+
+	// SourceCookie indicates the value is extracted from cookies.
+	SourceCookie
+
+	// SourceCustom indicates the value is extracted using a custom extractor function.
+	SourceCustom
+)
+
+// ErrNotFound is returned when the requested value is missing or empty.
+var ErrNotFound = errors.New("value not found")
+
+// Extractor defines a value extraction method with metadata.
+type Extractor struct {
+	Extract    func(fiber.Ctx) (string, error)
+	Key        string      // The parameter/header name used for extraction
+	AuthScheme string      // The auth scheme used, e.g., "Bearer"
+	Chain      []Extractor // For chained extractors, stores all extractors in the chain
+	Source     Source      // The type of source being extracted from
+}
+
+// FromAuthHeader extracts a value from the Authorization header with an optional prefix.
+// This function implements RFC 7235 compliant Authorization header parsing.
+//
+// The function supports:
+//   - Case-insensitive auth scheme matching
+//   - Empty auth scheme for raw header extraction
+//
+// Parameters:
+//   - authScheme: The auth scheme to strip from the header value (e.g., "Bearer", "Basic").
+//     If empty, the entire header value is returned without modification.
+//
+// Returns:
+//
+//	An Extractor that attempts to retrieve and parse the Authorization header.
+//	Returns ErrNotFound if the header is missing, malformed, or doesn't match the expected scheme.
+//
+// Examples:
+//
+//	// Extract Bearer token
+//	extractor := FromAuthHeader("Bearer")
+//	// Input: "Bearer abc123" -> Output: "abc123"
+//	// Input: "Basic dXNlcjpwYXNz" -> Output: ErrNotFound
+//
+//	// Extract raw header value
+//	extractor := FromAuthHeader("")
+//	// Input: "CustomAuth token123" -> Output: "CustomAuth token123"
+func FromAuthHeader(authScheme string) Extractor {
```

Comment:
We should copy the one from keyauth:

https://github.com/gofiber/fiber/blob/main/middleware/keyauth/extractors.go#L51

Reply:
I was going to change it in keyauth. FromAuthHeader already implies its Authorization. So having two parameters, while allowing flexibility, don’t make sense.